frontend/src/concepts/modelCatalog/useModelCatalogSources.ts

```diff
@@ -20,18 +20,34 @@ export const useModelCatalogSources = (): FetchState<State> => {
       throw new NotReadyError('Model catalog feature is not enabled');
     }
 
-    const configMap = await getConfigMap(dashboardNamespace, MODEL_CATALOG_SOURCE_CONFIGMAP);
-
-    if (!configMap.data?.modelCatalogSource) {
-      return [];
-    }
-
     try {
-      const source: ModelCatalogSource = JSON.parse(configMap.data.modelCatalogSource);
-      return [source];
-    } catch (e) {
-      // Swallow JSON parse errors and return empty array for temporary summit implementation
-      return [];
+      const configMap = await getConfigMap(dashboardNamespace, MODEL_CATALOG_SOURCE_CONFIGMAP);
+      /* eslint-enable @typescript-eslint/no-unsafe-member-access */
+
+      if (!configMap.data || !configMap.data.modelCatalogSource) {
+        return [];
+      }
+
+      try {
+        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+        const source = JSON.parse(configMap.data.modelCatalogSource) as ModelCatalogSource;
```

Comment:
You don't need this linter override here if you specify the type of source.
```suggestion
        const source: ModelCatalogSource = JSON.parse(configMap.data.modelCatalogSource);
```